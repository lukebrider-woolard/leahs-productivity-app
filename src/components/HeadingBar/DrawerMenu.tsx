import { useNavigate } from 'react-router-dom';

import { Drawer, List, ListItem, ListItemButton, ListItemIcon, ListItemText } from '@mui/material';
import DatasetOutlinedIcon from '@mui/icons-material/DatasetOutlined';
import HomeOutlinedIcon from '@mui/icons-material/HomeOutlined';

interface Props {
  isOpen: boolean;
  setOpen: (isOpen: boolean) => void;
}

export default function DrawerMenu(props: Props) {
  const navigate = useNavigate();

  const toggleDrawer =
  (open: boolean) =>
  (event: React.KeyboardEvent | React.MouseEvent) => {
    if (
      event.type === 'keydown' &&
      ((event as React.KeyboardEvent).key === 'Tab' ||
        (event as React.KeyboardEvent).key === 'Shift')
    ) {
      return;
    }

    props.setOpen(open);
  };

  return (
    <Drawer open={props.isOpen} onClose={toggleDrawer(false)}>
      <List>
        <ListItem key="Dashboard" disablePadding>
          <ListItemButton onClick={() => navigate('/')}>
            <ListItemIcon>
              <HomeOutlinedIcon color="secondary"/>
            </ListItemIcon>
            <ListItemText primary="Dashboard" />
          </ListItemButton>
        </ListItem>
        <ListItem key="DataPage" disablePadding>
          <ListItemButton onClick={() => navigate('/datapage')}>
            <ListItemIcon>
              <DatasetOutlinedIcon color="secondary"/>
            </ListItemIcon>
            <ListItemText primary="Magnets Data Display" />
          </ListItemButton>
        </ListItem>
      </List>
    </Drawer>
  )
}
