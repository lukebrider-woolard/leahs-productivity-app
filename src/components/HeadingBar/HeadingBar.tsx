import { useState } from 'react';

import { AppBar, Box, IconButton, Toolbar, Typography } from '@mui/material';
import MenuIcon from '@mui/icons-material/Menu';
import PetsIcon from '@mui/icons-material/Pets';

import DrawerMenu from './DrawerMenu';

export default function HeadingBar() {
  const [drawerOpen, setDrawerOpen] = useState<boolean>(false);

  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar position="static">
        <Toolbar>
          <IconButton
            size="large"
            edge="start"
            color="inherit"
            aria-label="menu"
            sx={{ mr: 2 }}
            onClick={() => setDrawerOpen(!drawerOpen)}
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
            Leah's Productivity App
          </Typography>
          <PetsIcon />
        </Toolbar>
      </AppBar>
      <DrawerMenu isOpen={drawerOpen} setOpen={setDrawerOpen} />
    </Box>
  )
};
