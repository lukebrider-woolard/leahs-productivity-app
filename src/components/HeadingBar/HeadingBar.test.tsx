import HeadingBar from './HeadingBar';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

beforeAll(() => {
  const mockedNavigate = jest.fn();

  jest.mock('react-router', () => ({
    ...jest.requireActual('react-router'),
    useNavigate: () => mockedNavigate,
  }));
});

afterAll(() => {
  jest.clearAllMocks();
});

describe('Heading Bar UI Tests', () => {
  test('heading bar renders as expected', () => {
    const HeadingBarTree = renderer
      .create(
        <BrowserRouter>
          <HeadingBar />
        </BrowserRouter>
      )
      .toJSON();

    expect(HeadingBarTree).toMatchSnapshot();
  });

  test('menu button opens drawer menu when clicked', async () => {
    render(
      <BrowserRouter>
        <HeadingBar />
      </BrowserRouter>
    );

    const menuButton = screen.getByRole('button', {
      name: /menu/i,
    });

    userEvent.click(menuButton);

    await waitFor(() => {
      expect(screen.getByRole('list')).toBeInTheDocument();
    });
  });
});
