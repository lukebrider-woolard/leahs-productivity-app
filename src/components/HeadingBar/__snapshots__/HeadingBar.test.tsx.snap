// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Bar UI Tests heading bar renders as expected 1`] = `
<div
  className="MuiBox-root css-i9gxme"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <button
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <div
        className="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
      >
        Leah's Productivity App
      </div>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="PetsIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <circle
          cx="4.5"
          cy="9.5"
          r="2.5"
        />
        <circle
          cx="9"
          cy="5.5"
          r="2.5"
        />
        <circle
          cx="15"
          cy="5.5"
          r="2.5"
        />
        <circle
          cx="19.5"
          cy="9.5"
          r="2.5"
        />
        <path
          d="M17.34 14.86c-.87-1.02-1.6-1.89-2.48-2.91-.46-.54-1.05-1.08-1.75-1.32-.11-.04-.22-.07-.33-.09-.25-.04-.52-.04-.78-.04s-.53 0-.79.05c-.11.02-.22.05-.33.09-.7.24-1.28.78-1.75 1.32-.87 1.02-1.6 1.89-2.48 2.91-1.31 1.31-2.92 2.76-2.62 4.79.29 1.02 1.02 2.03 2.33 2.32.73.15 3.06-.44 5.54-.44h.18c2.48 0 4.81.58 5.54.44 1.31-.29 2.04-1.31 2.33-2.32.31-2.04-1.3-3.49-2.61-4.8z"
        />
      </svg>
    </div>
  </header>
</div>
`;
